### DISCUSSION TITLE
Is there example of using [UseFiltering], [UseSorting], [UseOffsetPaging] with custom data source?

### DISCUSSION THREAD

**Discussion by ladislavmrnka** [ChilliCream/hotchocolate#6640] [Q&A]
Hello everyone,

we are trying to figure out how to use existing attributes for [UseFiltering], [UseSorting] and [UseOffsetPaging] to collect filtering, sorting and paging requirements and execute stored procedure or make HTTP call to another internal non-GraphQL service. While we can do this easily by giving up on existing attributes, creating our custom inputs and execute everything manually we got ourselves into bad situation where some of our existing API has been built with EF / IQueryable and these attributes. Now we need to add new features which cannot use EF but we need to keep the schema consistent.

Any advice welcome! Thank you.

**Comment by glen-84** [comment]
See https://github.com/ChilliCream/graphql-platform/issues/6676.
